consensus/src/experimental/buffer_item.rs

```diff
@@ -154,18 +195,54 @@ impl BufferItem {
                         commit_proof,
                         callback,
                     }))
-                } else {
-                    debug!(
-                        "{} advance to executed from ordered",
-                        commit_proof.commit_info()
+                } else if validator
+                    .check_voting_power(unverified_signatures.signatures().keys())
+                    .is_ok()
+                {
+                    let commit_ledger_info =
+                        generate_commit_ledger_info(&commit_info, &ordered_proof);
+
+                    let verified_signatures = get_verified_signatures(
+                        &unverified_signatures,
+                        validator,
+                        &commit_ledger_info,
                     );
-                    Self::Executed(Box::new(ExecutedItem {
+                    // We need to check for voting power again as some of the signatures might be
+                    // filtered out in the above step and the voting power might not be sufficient
+                    if (validator.check_voting_power(verified_signatures.signatures().keys()))
+                        .is_ok()
```

Comment:
this section is complicated, I think we could just do
1. get verified signatures
2. check voting power
3. executed item or aggregated item

Reply:
Yes, that is true but the reason I am not doing that is because, signature verification is expensive and we don't want to verify the signature before we have enough voting power.  Given the signature verification is in the critical path, I feel this is worth the complexity. Thoughts?